File: Coding-Interviews/16-数值的整数次方.cpp
#include <cmath>
using namespace std;
class Solution
{
public:
    double myPow(double x, int n)
    {
        if (n == 0)
            return 1;
        bool flag = false;
        if (n < 0) {
            n    = abs(n);
            flag = true;
        }
        double ret  = 1;
        auto   temp = myPow(x, n / 2);
        if (n % 2 == 1) {
            ret = x;
        }
        ret *= temp * temp;
        return flag ? 1 / ret : ret;
    }
};

File: Coding-Interviews/16-数值的整数次方.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0:
            return 1
        if n == 1:
            return x
        if n < 0:
            n = -n
            x = 1/x
        half = self.myPow(x, n//2)
        if n % 2:
            return half*half*x
        return half*half
